File: c++ programming task/login and registration system/Task2_LoginRegistration.java
import java.util.Scanner;
import java.io.*;

public class Task2_LoginRegistration {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        System.out.println("1. Register\n2. Login\nChoose:");
        int choice = sc.nextInt();
        sc.nextLine();
        if (choice == 1) register(sc);
        else login(sc);
    }

    static void register(Scanner sc) throws IOException {
        System.out.print("Username: ");
        String user = sc.nextLine();
        File file = new File(user + ".txt");
        if (file.exists()) {
            System.out.println("User exists.");
            return;
        }
        System.out.print("Password: ");
        String pass = sc.nextLine();
        FileWriter writer = new FileWriter(file);
        writer.write(user + "\n" + pass);
        writer.close();
        System.out.println("Registered!");
    }

    static void login(Scanner sc) throws IOException {
        System.out.print("Username: ");
        String user = sc.nextLine();
        File file = new File(user + ".txt");
        if (!file.exists()) {
            System.out.println("User not found.");
            return;
        }
        BufferedReader br = new BufferedReader(new FileReader(file));
        String storedUser = br.readLine();
        String storedPass = br.readLine();
        br.close();
        System.out.print("Password: ");
        String pass = sc.nextLine();
        if (storedPass.equals(pass))
            System.out.println("Login successful!");
        else
            System.out.println("Login failed.");
    }
}

File: c++ programming task/login and registration system/task2_login_registration.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

bool userExists(const string &username) {
    ifstream file(username + ".txt");
    return file.good();
}

void registerUser() {
    string username, password;
    cout << "Register
Username: ";
    cin >> username;
    if (userExists(username)) {
        cout << "Error: Username already exists.\n";
        return;
    }
    cout << "Password: ";
    cin >> password;
    ofstream file(username + ".txt");
    file << username << endl << password << endl;
    file.close();
    cout << "Registration successful!\n";
}

void loginUser() {
    string username, password, storedUser, storedPass;
    cout << "Login\nUsername: ";
    cin >> username;
    cout << "Password: ";
    cin >> password;
    ifstream file(username + ".txt");
    if (!file) {
        cout << "Error: User not found.\n";
        return;
    }
    getline(file, storedUser);
    getline(file, storedPass);
    file.close();
    if (username == storedUser && password == storedPass)
        cout << "Login successful!\n";
    else
        cout << "Login failed. Incorrect password.\n";
}

int main() {
    int choice;
    cout << "1. Register\n2. Login\nChoice: ";
    cin >> choice;
    if (choice == 1) registerUser();
    else if (choice == 2) loginUser();
    else cout << "Invalid choice.\n";
    return 0;
}
